src/home/home.dao.js

```diff
@@ -0,0 +1,103 @@
+import { BaseError } from "../../config/error.js";
+import { status } from "../../config/response.status.js";
+import { pool } from "../../config/db.config.js";
+import { getShortsByCategory,getAllCategories, getFollowerFeed, getShort, getUserCategories, getUserRecommendedShorts } from "./home.sql.js";
+
+// 카테고리 별 쇼츠 조회
+export const getShortsbyCategory = async (category_id, offset, limit) => {
+    try {
+        const conn = await pool.getConnection();
+        const [categoryShorts] = await pool.query(getShortsByCategory, [category_id, limit, offset]);
+
+        conn.release(); 
+
+        return categoryShorts;
+    }
+    catch(err){
+        throw new BaseError(status.BAD_REQUEST);
+    }
+}
+
+// 비회원일 시 카테고리 리스트 반환 - 우선순위순
+export const getAllCategory = async () => {
+    try {
+        const conn = await pool.getConnection();
+
+        let allCategories = []
+        [allCategories] = await conn.query(getAllCategories);
```

Comment:
여기서 RefferenceError가 발생하고 있습니다!
```shell
ReferenceError: Cannot access 'allCategories' before initialization
    at getAllCategory (file:///Users/ossu-dev/umc/Read_Me_Node.js/src/home/home.dao.js:27:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getMainInfo (file:///Users/ossu-dev/umc/Read_Me_Node.js/src/home/home.service.js:26:7)
    at async getHomeInfo (file:///Users/ossu-dev/umc/Read_Me_Node.js/src/home/home.controller.js:27:20)
```

아마 let 선언 후 `;` 세미콜론이 누락되어 발생한 에러 같은데 이 코드 상에서 `allCategories`가 변수일 필요가 없어서 아래와 같이 수정하면 될 것 같아요!
```suggestion
        const [allCategories] = await conn.query(getAllCategories);
```
```